tests/typecheck/conf/test_settings.yml

```diff
@@ -15,7 +15,7 @@
 
                 def test() -> None:
                     reveal_type(settings.IMMEDIATE_VALUE)        # N: Revealed type is "builtins.int"
-                    reveal_type(settings.CIRCULAR_WITHOUT_HINT)  # E: Import cycle from Django settings module prevents type inference for 'CIRCULAR_WITHOUT_HINT'  [misc]  # N: Revealed type is "Any"
+                    reveal_type(settings.CIRCULAR_WITHOUT_HINT)  # N: Revealed type is "builtins.int"
```

Comment:
I wonder if we can come up with another test case that produces the "Import cycle from Django settings module prevents type inference" error?

Would be nice to have test coverage for this code still.

Reply:
Yeah, agreed but I'm not sure what that test could look like either.

If I've read the plugin code correctly it only interferes when the type checking haven't inferred a type of the settings module at the stage where plugin is invoked.

I tried importing like `import django.conf`, since that seemed to have lower priority. Then I get the error, but I'm not sure if that could be because `get_additional_deps` doesn't consider that style of importing properly.

Reply:
Indeed, updated the testcase as you suggested, thanks!